utils/notifications.js

```diff
@@ -29,25 +29,35 @@ async function setupTwilioSms() {
   return [twilioClient, fetchedSecrets.messagingServiceSid];
 }
 
-const sendSmsBatch = async (smsDataArray) => {
+const sendSmsBatch = async (smsRecordArray) => {
   // TODO: further check to see whether 1000 is a good batch size
-  const chunkArray = createChunkArray(smsDataArray, 1000);
+  let adjustedDataArray = [];
+  const chunkArray = createChunkArray(smsRecordArray, 1000);
   for (const chunk of chunkArray) {
     const messageArray = await Promise.all(
```

Comment:
Promise.all will resolve when the first error occurs and will not wait for the results of the rest of the promises.  I would recommend using Promise.allSettled instead.